src/Nethermind/Nethermind.Network.Stats/Model/InitiateDisconnectReason.cs

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Stats.Model;
+
+/// <summary>
+/// Nethermind level disconnect reason. Don't forget to add the corresponding Eth level disconnect reason in `InitiateDisconnectReasonExtension`.
+/// </summary>
+public enum InitiateDisconnectReason : byte
+{
+    IncomingConnectionRejectedTooManyPeer,
+    SessionAlreadyExist,
+    LostSessionDirectionDecision,
+    OppositeDirectionCleanup,
+
+    SnapServerNotImplemented,
+    IncompatibleP2PVersion,
+    InvalidCapability,
+    InvalidChainId,
+    InvalidGenesis,
+    ProtocolInitTimeout,
```

Comment:
Just `Timeout`/`InitTimeout`?

Reply:
No, I think this one is accurate. Might want to differentiate with other timeout in future.